rules/CodeQuality/Rector/FuncCall/ChangeArrayPushToArrayAssignRector.php

```diff
@@ -59,61 +46,54 @@ public function run()
      */
     public function getNodeTypes(): array
     {
-        return [FuncCall::class];
+        return [Expression::class];
     }
 
     /**
-     * @param FuncCall $node
+     * @param Expression[] $node
+     * @param Expression[]|null $node
      */
-    public function refactor(Node $node): ?Node
+    public function refactor(Node $node): ?array
     {
-        if (! $this->isName($node, 'array_push')) {
+        if (! $node->expr instanceof FuncCall) {
             return null;
         }
 
-        if ($this->hasArraySpread($node)) {
+        $funcCall = $node->expr;
+        if (! $this->isName($funcCall, 'array_push')) {
             return null;
         }
 
-        $parent = $node->getAttribute(AttributeKey::PARENT_NODE);
-        if (! $parent instanceof Expression) {
+        if ($this->hasArraySpread($funcCall)) {
             return null;
         }
 
-        if (! isset($node->args[0])) {
-            return null;
-        }
+        $args = $funcCall->getArgs();
```

Comment:
`$args === []` need to be checked early, return null

Reply:
@TomasVotruba empty args is allowed while show warning in php 7.4, ref https://3v4l.org/TEJCb#v740 so empty args needs to be checked.

Reply:
I see. Could you fix it in a PR?

Reply:
I will try, also array_push allow to not having 2nd arg https://3v4l.org/YTXan

Reply:
https://github.com/rectorphp/rector-src/pull/2266